src/components/core/check-overflow/index.js

```diff
@@ -3,9 +3,9 @@ function checkOverflow() {
   const swiper = this;
   const params = swiper.params;
   const wasLocked = swiper.isLocked;
-  const lastSlidePosition = (params.slidesOffsetBefore + (params.spaceBetween * (swiper.slides.length - 1)) + ((swiper.slides[0]).offsetWidth) * swiper.slides.length);
+  const lastSlidePosition = swiper.slides.length > 0 ? (params.slidesOffsetBefore + (params.spaceBetween * (swiper.slides.length - 1)) + ((swiper.slides[0]).offsetWidth) * swiper.slides.length) : undefined;
 
-  if ((typeof params.slidesOffsetBefore !== 'undefined' && params.slidesOffsetBefore !== 0) || (typeof params.slidesOffsetAfter !== 'undefined' && params.slidesOffsetAfter !== 0)) {
+  if (((typeof params.slidesOffsetBefore !== 'undefined' && params.slidesOffsetBefore !== 0) || (typeof params.slidesOffsetAfter !== 'undefined' && params.slidesOffsetAfter !== 0)) && lastSlidePosition !== 'undefined') {
```

Comment:
You could shorten this as well since the if statement would catch both undefined and 0 👍 
```
if(params.slidesOffsetBefore && params.slidesOffsetAfter && lastSlidePosition) {
...
}
```

Reply:
Clever shorthands, I like it. Thanks!